core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -379,6 +379,41 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     }
   executorAllocationManager.foreach(_.start())
 
+  // Use reflection to instantiate listeners specified via the `spark.extraListeners` configuration
+  // or the SPARK_EXTRA_LISTENERS environment variable
+  try {
+    val listenerClassNames: Seq[String] = {
+      val fromSparkConf = conf.get("spark.extraListeners", "").split(',')
+      val fromEnvVar = Option(conf.getenv("SPARK_EXTRA_LISTENERS")).getOrElse("").split(',')
```

Comment:
This functions slightly differently than the rest of our env-based configurations: the settings from this configuration are merged with the SparkConf configuration, rather than overriding/being-overriden by it.  My motivation for this configuration was to provide a mechanism for the execution environment to inject custom listeners without having to worry about them being overriden by settings in the user's SparkConf (this is necessary because we don't have a mechanism for automatic merging of Spark configurations).